packages/react-dom/src/__tests__/ReactDOMFizzServer-test.js

```diff
@@ -1099,6 +1099,85 @@ describe('ReactDOMFizzServer', () => {
     expect(loggedErrors).toEqual([theError]);
   });
 
+  fit('Errors in boundaries should not be logged when new, unrelated Suspense boundaries commit', async () => {
+    class ErrorBoundary extends React.Component {
+      state = {error: null};
+      static getDerivedStateFromError(error) {
+        return {error};
+      }
+
+      render() {
+        if (this.state.error) {
+          return <Text text="did error" />;
+        }
+        return this.props.children;
+      }
+    }
+
+    const theError = new Error('Boom');
+    function Page() {
+      throw theError;
+    }
+
+    let renderSuspense;
+    function Sibling() {
+      const [state, setState] = React.useState(false);
+      renderSuspense = () => setState(true);
+
+      return state ? <Suspense /> : null;
+    }
+
+    function App({isClient}) {
+      return (
+        <>
+          <ErrorBoundary>
+            <Suspense>
+              <Page />
+            </Suspense>
+          </ErrorBoundary>
+          <Sibling />
+        </>
+      );
+    }
+
+    const ssrErrors = [];
+    function onError(x) {
+      ssrErrors.push(x);
+      return 'hash(' + x.message + ')';
+    }
+    const expectedDigest = onError(theError);
+    ssrErrors.length = 0;
+
+    await act(() => {
+      const {pipe} = renderToPipeableStream(
+        <App />,
+
+        {
+          onError,
+        },
+      );
+      pipe(writable);
+    });
+    expect(ssrErrors).toEqual([theError]);
+
+    const errors = [];
+    ReactDOMClient.hydrateRoot(container, <App />, {
+      onRecoverableError(error, errorInfo) {
+        errors.push({error, errorInfo});
+      },
+    });
+    await waitForAll([]);
+
+    expect(getVisibleChildren(container)).toEqual('did error');
+    expectErrors(errors, [], []);
+
+    renderSuspense();
+    await waitForAll([]);
+
+    expect(getVisibleChildren(container)).toEqual('did error');
+    expectErrors(errors, [], []);
```

Comment:
There should be no errors here yet React is resurfacing a recoverable error from an unrelated tree.